app/src/org/commcare/preferences/CommCarePreferences.java

```diff
@@ -79,85 +105,137 @@
-        lp.setEntries(ChangeLocaleUtil.getLocaleNames());
-        lp.setEntryValues(ChangeLocaleUtil.getLocaleCodes());
-        lp.setTitle(Localization.get("home.menu.locale.change"));
-        lp.setKey(PREFS_LOCALE_KEY);
-        lp.setDialogTitle(Localization.get("home.menu.locale.select"));
-        this.getPreferenceScreen().addPreference(lp);
-        updatePreferencesText();
-        setTitle("CommCare" + " > " + "Application Preferences");
-
-        populatePrefKeyToEventLabelMapping();
+        setupUI();
+
         GoogleAnalyticsUtils.createPreferenceOnClickListeners(prefMgr, prefKeyToAnalyticsEvent,
                 GoogleAnalyticsFields.CATEGORY_CC_PREFS);
         // Override the default listener for the print pref key b/c it has extra behavior
         createPrintPrefOnClickListener(prefMgr);
     }
 
-    private static void populatePrefKeyToEventLabelMapping() {
-        prefKeyToAnalyticsEvent.put(PREFS_APP_SERVER_KEY, GoogleAnalyticsFields.LABEL_APP_SERVER);
-        prefKeyToAnalyticsEvent.put(PREFS_DATA_SERVER_KEY, GoogleAnalyticsFields.LABEL_DATA_SERVER);
-        prefKeyToAnalyticsEvent.put(PREFS_SUBMISSION_URL_KEY, GoogleAnalyticsFields.LABEL_SUBMISSION_SERVER);
-        prefKeyToAnalyticsEvent.put(PREFS_KEY_SERVER_KEY, GoogleAnalyticsFields.LABEL_KEY_SERVER);
-        prefKeyToAnalyticsEvent.put(PREFS_FORM_RECORD_KEY, GoogleAnalyticsFields.LABEL_FORM_RECORD_SERVER);
-        prefKeyToAnalyticsEvent.put(AUTO_UPDATE_FREQUENCY, GoogleAnalyticsFields.LABEL_AUTO_UPDATE);
-        prefKeyToAnalyticsEvent.put(PREFS_FUZZY_SEARCH_KEY, GoogleAnalyticsFields.LABEL_FUZZY_SEARCH);
-        prefKeyToAnalyticsEvent.put(PREFS_LOCALE_KEY, GoogleAnalyticsFields.LABEL_LOCALE);
-        prefKeyToAnalyticsEvent.put(PREFS_PRINT_DOC_LOCATION, GoogleAnalyticsFields.LABEL_PRINT_TEMPLATE);
-        prefKeyToAnalyticsEvent.put(GRID_MENUS_ENABLED, GoogleAnalyticsFields.LABEL_GRID_MENUS);
+    private void setupUI() {
+        setTitle(Localization.get("settings.main.title"));
+        addBackButtonToActionBar(this);
+        setupLocalizedText(this, keyToTitleMap);
+        setupButtons();
+    }
+
+    public static void setupLocalizedText(PreferenceActivity activity,
+                                             Map<String, String> prefToTitleMap) {
+        PreferenceScreen screen = activity.getPreferenceScreen();
+        for (int i = 0; i < screen.getPreferenceCount(); i++) {
+            String key = screen.getPreference(i).getKey();
+            if (prefToTitleMap.containsKey(key)) {
+                try {
+                    String localizedString = Localization.get(prefToTitleMap.get(key));
+                    screen.getPreference(i).setTitle(localizedString);
+                } catch (NoLocalizedTextException nle) {
+                    Log.w(TAG, "Unable to localize: " + prefToTitleMap.get(key));
+                }
+            }
+        }
+    }
+
+    public static void addBackButtonToActionBar(Activity activity) {
```

Comment:
Just and improvement that doesn't have to do with other code changes. Adds a `<` to the action bar nav. If the back button on a device is broken (has happened to me before) then getting out of menus without the `<` is near impossible.